File: Array/GoLangDsa/SlidingWindow/Leetcode1248.go
package main

import "fmt"

func numberOfSubarrays(nums []int, k int) int {

	Left := 0
	Right := 0
	Length := 0
	maxLen := 0

	for Right < len(nums) {

		if Right-Left+1 == 1 {

			Right++
			continue
		}

		l, r := Left, Right
		andResult := nums[l]
		storeAnd := []int{}
		for l <= r {

			andResult &= nums[l]
			storeAnd = append(storeAnd, andResult)
			l++
		}

		sum := 0
		for i := 0; i < len(storeAnd); i++ {

			sum += storeAnd[i]
		}

		if sum == 0 {

			Length = Right - Left + 1
			if Length > maxLen {
				maxLen = Length
			}

		} else {

			Left++
		}

		Right++

	}

	return maxLen

}

File: Array/Leetcode1248.cpp
#include<vector>

int atMostKOddNumbers(const std::vector<int>& nums, int k) 
{

        int n = nums.size();
        int Initialptr{0};
        int Endptr{0};
        int len{0};
        int sum{0};
        int count{0};

        if (k < 0) 
        {
            return 0;
        }

        while (Endptr < n) 
        {
            sum += nums[Endptr] % 2;
            while (sum > k) 
            {
                sum -= nums[Initialptr] % 2;
                ++Initialptr;
            }
            len = Endptr - Initialptr + 1;
            count += len;
            ++Endptr;
        }
        return count;
}

    int numberOfSubarrays(std::vector<int>& nums, int k) 
    {
        return atMostKOddNumbers(nums, k) - atMostKOddNumbers(nums, k - 1);
    }